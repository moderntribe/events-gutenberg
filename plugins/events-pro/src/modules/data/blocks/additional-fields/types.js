/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const ADD_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/ADD_ADDITIONAL_FIELD`;
export const REMOVE_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD`;

export const SET_ADDITIONAL_FIELD_LABEL = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_LABEL`;
export const SET_ADDITIONAL_FIELD_IS_PRISTINE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_IS_PRISTINE`;
export const SET_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_TYPE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_TYPE`;
export const SET_ADDITIONAL_FIELD_OPTIONS = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OPTIONS`;
export const SET_ADDITIONAL_FIELD_DIVIDER_LIST = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_LIST`;
export const SET_ADDITIONAL_FIELD_DIVIDER_END = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_END`;
