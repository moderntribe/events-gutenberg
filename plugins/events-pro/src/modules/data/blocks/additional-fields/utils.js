export const FIELD_TYPES = {
	CHECKBOX: 'CHECKBOX',
	TEXT: 'TEXT',
	DROP_DOWN: 'DROP_DOWN',
	URL: 'URL',
	RADIO: 'RADIO',
	TEXT_AREA: 'TEXT_AREA',
};
