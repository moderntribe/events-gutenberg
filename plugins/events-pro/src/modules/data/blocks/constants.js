//
// ─── BLOCK TYPES ────────────────────────────────────────────────────────────────
//


export const RECURRING = 'recurring';
export const EXCEPTION = 'exception';

export const BLOCK_TYPES = [ RECURRING, EXCEPTION ];

//
// ─── STATE KEYS ─────────────────────────────────────────────────────────────────
//

export const KEY_TYPE = 'type';
export const KEY_ALL_DAY = 'all_day';
export const KEY_MULTI_DAY = 'multi_day';
export const KEY_START_TIME = 'start_time';
export const KEY_END_TIME = 'end_time';
export const KEY_START_DATE = 'start_date';
export const KEY_LIMIT = 'limit';
export const KEY_LIMIT_TYPE = 'limit_type';
export const KEY_BETWEEN = 'between';
export const KEY_TIMEZONE = 'timezone';
