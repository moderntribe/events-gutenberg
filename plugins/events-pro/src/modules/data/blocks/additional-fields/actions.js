/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

