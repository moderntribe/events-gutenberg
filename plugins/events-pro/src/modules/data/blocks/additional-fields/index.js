/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as utils from './utils';

export default reducer;
export { types, actions, selectors, utils };
