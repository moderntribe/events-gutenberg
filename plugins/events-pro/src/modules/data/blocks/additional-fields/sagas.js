/**
 * External dependencies
 */
import { takeEvery, put, select, all } from 'redux-saga/effects';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';
import identity from 'lodash/identity';

/**
 * Internal dependencies
 */
import { actions, types, selectors } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';

export function* setInitialState( props ) {
	const { payload = {} } = props;
	const { name = '', get = noop } = payload;
	const type = get( 'type' );
	const value = get( 'value' );

	yield all( [
		put( actions.addField( name ) ),
		put( actions.setFieldIsPristine( name, get( 'isPristine' ) ) ),
		put( actions.setFieldType( name, type ) ),
		put( actions.setFieldLabel( name, get( 'label' ) ) ),
		put( actions.setFieldOptions( name, get( 'options' ) ) ),
		put( actions.setFieldValue( name, get( 'value' ) ) ),
	] );
}

export function* setPristineState( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value  = yield select( selectors.getFieldValue, payload );
	const isPristine = value === null || '' === value || isEmpty( value );
	yield put( actions.setFieldIsPristine( name, isPristine ) );
}

export function* appendFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = [ ...current, value ]
		.filter( identity )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* removeFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = current
		.filter( ( text ) => text !== value )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}


export default function* watchers() {
	yield takeEvery( types.SET_ADDITIONAL_FIELD_INITIAL_STATE, setInitialState );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_CHANGE, setPristineState );
	yield takeEvery( types.APPEND_ADDITIONAL_FIELD_VALUE, appendFieldValue );
	yield takeEvery( types.REMOVE_ADDITIONAL_FIELD_VALUE, removeFieldValue );
}
