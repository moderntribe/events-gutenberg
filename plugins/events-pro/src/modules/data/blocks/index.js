/**
 * Internal dependencies
 */
import * as constants from './constants';
import reducer from './reducer';
import './sagas';

export default reducer;
export { constants };
