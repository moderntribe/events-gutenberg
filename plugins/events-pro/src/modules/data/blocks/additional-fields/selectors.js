/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);
export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);
