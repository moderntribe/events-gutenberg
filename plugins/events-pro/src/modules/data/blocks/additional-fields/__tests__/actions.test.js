/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';

describe( 'Actions in Additional Fields', () => {
	test( 'add field', () => {
		expect( actions.addField( 'Host' ) ).toMatchSnapshot();
	} );

	test( 'remove field', () => {
		expect( actions.removeField( 'Host' ) ).toMatchSnapshot();
	} );

	test( 'set field name', () => {
		expect( actions.setFieldName( 'Host' ) ).toMatchSnapshot();
	} );

	test( 'set field is pristine', () => {
		expect( actions.setFieldIsPristine( 'Host', true ) ).toMatchSnapshot();
		expect( actions.setFieldIsPristine( 'Host', false ) ).toMatchSnapshot();
	} );

	test( 'set field value', () => {
		expect( actions.setFieldValue( 'Host', 'Rachel MA' ) ).toMatchSnapshot();
	} );

	test( 'set field type', () => {
		expect( actions.setFieldType( 'Host', FIELD_TYPES.checkbox ) ).toMatchSnapshot();
	} );

	test( 'set field options', () => {
		expect( actions.setFieldOptions( 'Host', [ 'Option 1', 'Option 2 ' ] ) ).toMatchSnapshot();
	} );

	test( 'set field list divider', () => {
		expect( actions.setFieldDividerList( 'Host', ', ' ) ).toMatchSnapshot();
	} );

	test( 'set field divider end', () => {
		expect( actions.setFieldDividerEnd( 'Host', ' & ' ) ).toMatchSnapshot();
	} );
} );
