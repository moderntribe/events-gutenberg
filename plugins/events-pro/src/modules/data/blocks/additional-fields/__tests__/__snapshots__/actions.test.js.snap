// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions in Additional Fields add field 1`] = `
Object {
  "payload": Object {
    "name": "Host",
  },
  "type": "@@MT/EVENTS-PRO/ADD_ADDITIONAL_FIELD",
}
`;

exports[`Actions in Additional Fields remove field 1`] = `
Object {
  "payload": Object {
    "name": "Host",
  },
  "type": "@@MT/EVENTS-PRO/REMOVE_ADDITIONAL_FIELD",
}
`;

exports[`Actions in Additional Fields set field divider end 1`] = `
Object {
  "payload": Object {
    "dividerEnd": " & ",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_DIVIDER_END",
}
`;

exports[`Actions in Additional Fields set field is pristine 1`] = `
Object {
  "payload": Object {
    "isPristine": true,
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_IS_PRISTINE",
}
`;

exports[`Actions in Additional Fields set field is pristine 2`] = `
Object {
  "payload": Object {
    "isPristine": false,
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_IS_PRISTINE",
}
`;

exports[`Actions in Additional Fields set field list divider 1`] = `
Object {
  "payload": Object {
    "dividerList": ", ",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_DIVIDER_LIST",
}
`;

exports[`Actions in Additional Fields set field name 1`] = `
Object {
  "payload": Object {
    "name": "Host",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_NAME",
}
`;

exports[`Actions in Additional Fields set field options 1`] = `
Object {
  "payload": Object {
    "options": Array [
      "Option 1",
      "Option 2 ",
    ],
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_OPTIONS",
}
`;

exports[`Actions in Additional Fields set field type 1`] = `
Object {
  "payload": Object {
    "type": "checkbox",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_TYPE",
}
`;

exports[`Actions in Additional Fields set field value 1`] = `
Object {
  "payload": Object {
    "value": "Rachel MA",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_VALUE",
}
`;
